cpp/WriterPropertiesBuilder.cpp

```diff
@@ -209,4 +209,21 @@ extern "C"
 	{
 		TRYCATCH(builder->disable_page_checksum();)
 	}
+
+	PARQUETSHARP_EXPORT ExceptionInfo* WriterPropertiesBuilder_Sorting_Columns(WriterProperties::Builder* builder, int32_t* column_indices, bool* descending, bool* nulls_first, int num_columns)
+	{
+		std::vector<parquet::SortingColumn> sorting_columns;
+		
```

Comment:
We know exactly how many elements there will be in the vector, so we can reserve necessary storage to avoid reallocations:
```suggestion
		sorting_columns.reserve(num_columns);
```